resident/resident-service/src/main/java/io/mosip/resident/service/impl/ProxyMasterdataServiceImpl.java

```diff
@@ -413,5 +462,118 @@ public ResponseWrapper<?> getGenderTypesByLangCode(String langCode) throws Resid
 		logger.debug("ProxyMasterdataServiceImpl::getGenderTypesByLangCode()::exit");
 		return responseWrapper;
 	}
+  
+	/**
+	 * download registration centers based on language code, hierarchyLevel and center names
+	 */
+	public InputStream downloadRegistrationCentersByHierarchyLevel(String langCode, Short hierarchyLevel,
```

Comment:
This method should not be in the proxy. It should be in a independent service.

Reply:
Define the new controller and Service classes